model/distro/db.go

```diff
@@ -140,12 +140,20 @@ func BySpawnAllowed() db.Q {
 	return db.Query(bson.M{SpawnAllowedKey: true})
 }
 
-// ByActiveOrStatic returns a query that selects only active or static distros
-func ByActiveOrStatic() db.Q {
-	return db.Query(bson.M{"$or": []bson.M{
-		bson.M{DisabledKey: bson.M{"$exists": false}},
-		bson.M{ProviderKey: evergreen.HostTypeStatic},
-	}})
+// ByNeedsPlanning returns a query that selects only active or static distros that don't run containers
+func ByNeedsPlanning(containerPools []evergreen.ContainerPool) db.Q {
+	poolDistros := []string{}
+	for _, pool := range containerPools {
+		poolDistros = append(poolDistros, pool.Distro)
+	}
+	return db.Query(bson.M{
+		"_id": bson.M{
+			"$nin": poolDistros,
+		},
+		"$or": []bson.M{
+			bson.M{DisabledKey: bson.M{"$exists": false}},
```

Comment:
what about exists and is false

Reply:
This part of the query was unchanged. The field is omitempty, so I think the only scenario where that can happen is if the data is corrupt

Reply:
or presuamby if someone sets it to true? maybe the update uses $unset, but that seems scary.

Reply:
The update does an update with the entire doc so that should be fine